docs/adr/0002-yarn-v4.md

```diff
@@ -0,0 +1,243 @@
+- plugin-dlx: install a package in temporary environment
+- plugin-github: improves the performance when cloning from Github
+- plugin-nm: support for installing packages in `node_modules`
+- plugin-pnp: support for [Plug'n'Play](https://yarnpkg.com/features/pnp)
+- plugin-pnpm: support for installing packages using symlinks
+- plugin-typescript: Automatically adds `@types/` packages into your dependencies
+</details>
+<br>
+
+We should still be able to proceed only by disabling non-official plugins without introducing any arbitrary code execution, and keep the current behavior for v3 projects unchanged.
+
+#### A note about plugin-typescript
+
+The [typescript](https://yarnpkg.com/api/plugin-typescript) plugin is used to automatically include types when adding a dependency that does not package them by default. Since these changes are reflected in the `package.json` file, Cachi2 will handle them normally.
+
+<details>
+    <summary>Example of how the Typescript plugin works</summary>
+
+```
+$ yarn add lodash 
+➤ YN0000: · Yarn 4.5.3
+➤ YN0000: ┌ Resolution step
+➤ YN0085: │ + @types/lodash@npm:4.17.13, lodash@npm:4.17.21
+➤ YN0000: └ Completed
+➤ YN0000: ┌ Fetch step
+➤ YN0013: │ A package was added to the project (+ 957.26 KiB).
+➤ YN0000: └ Completed in 0s 252ms
+➤ YN0000: ┌ Link step
+➤ YN0000: └ Completed
+➤ YN0000: · Done in 0s 313ms
+
+$ cat package.json 
+{
+  "name": "yarn-types",
+  "packageManager": "yarn@4.5.3",
+  "dependencies": {
+    "lodash": "^4.17.21"
+  },
+  "devDependencies": {
+    "@types/lodash": "^4"
+  }
+}
+```
+</details>
+
+### pnpDataPath is no longer configurable
+
+This used to be a configurable key in the `.yarnrc.yml` file. The previous default path, `./.pnp.data.json`, is now hard-coded and can't be changed.
+
+The only mention to `pnpDataPath` in Cachi2 is the [check](https://github.com/containerbuildsystem/cachi2/blob/a5f19c6f9be90be4289beee35ecccd2827bbb328/cachi2/core/package_managers/yarn/main.py#L43) for paths pointing outside of the repo. The check can be kept the same way so Yarn v3 can still be covered, and it will simply be skipped in v4.
+
+### Yarn now caches npm version metadata
+
+Although Yarn only seems to create the cache folder (`{globalFolder}/metadata/npm`) when the [hardened mode](https://yarnpkg.com/configuration/yarnrc#enableHardenedMode) is enabled. Simply running `yarn install` on a local project or even prefetching with Cachi2 doesn't seem to generate this extra folder.
+
+The metadata, in itself, is a collection of json files with a few kilobytes each. In case we decide to enable the hardened mode, it'd be ideal to delete this npm metadata this from the output folder, although there's no harm in keeping it.
+
+### Changes in yarn.lock
+
+When updating v3 projects to v4, some differences in the lockfile appear. These are some picks from updating the [disallowed-protocols]() branch on the cachi2-yarn-berry test project:
```

Comment:
`disallowed-protocols` link appears to be broken.

Reply:
Fixed.